src/workerd/io/frankenvalue.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/io/frankenvalue.capnp.h>
+#include <workerd/jsg/jsg.h>
+
+namespace workerd {
+
+class Frankenvalue {
+ public:
+  Frankenvalue(): value(EmptyObject()) {}
+
+  bool empty() const {
+    return value.is<EmptyObject>() && properties.empty();
+  }
+
+  Frankenvalue clone() const;
+
+  // Convert to/from capnp format.
+  void toCapnp(rpc::Frankenvalue::Builder builder) const;
+  static Frankenvalue fromCapnp(rpc::Frankenvalue::Reader reader);
+
+  // Convert to/from JavaScript values. Note that round trips here don't produce the exact same
+  // Frankenvalue representation: toJs() puts all the contents together into a single value, and
+  // fromJs() always returns a Frakenvalue containing a single V8-serialized value.
+  jsg::JsValue toJs(jsg::Lock& js) const;
+  static Frankenvalue fromJs(jsg::Lock& js, jsg::JsValue value);
```

Comment:
non-blocking nit: If converting to Json is going to be a common use case, then maybe consider exposing a `jsg::JsString toJson(jsg::Lock& js) const;` also?

Reply:
No, you generally never convert a Frakenvalue to JSON. Parts of the Frakenvalue might come _from_ JSON (e.g. the binding configuration), but other parts may come from other places and may or may not be JSON-serializable. The final destination of a Frakenvalue is always a JS value in the callee isolate.